File: fix_hw.js
const fs = require('fs');
let content = fs.readFileSync('index.html', 'utf8');

// ── PATCH 1: 헤더 색상 원래대로 되돌리기 ──────────────────────
const wrongCard =
  '        <div class="prog-group-card" style="border-left:4px solid #3b82f6; margin-bottom:12px">\r\n' +
  '          <div class="prog-group-header" style="background:#eff6ff; padding:12px 14px; border-radius:12px 12px 0 0">\r\n' +
  '            <div class="prog-group-title" style="color:#2563eb; font-size:16px">${label}${subjectBadge}\r\n' +
  '              ${headerDateStr}\r\n' +
  '            </div>\r\n' +
  '          </div>';

const origCard =
  '        <div class="prog-group-card" style="border-left:4px solid ${gc.color}; margin-bottom:12px">\r\n' +
  '          <div class="prog-group-header" style="background:${gc.bg}; padding:12px 14px; border-radius:12px 12px 0 0">\r\n' +
  '            <div class="prog-group-title" style="color:${gc.color}; font-size:16px">${label}${subjectBadge}\r\n' +
  '              ${headerDateStr}\r\n' +
  '            </div>\r\n' +
  '          </div>';

if (content.includes(wrongCard)) {
  content = content.replace(wrongCard, origCard);
  console.log('PATCH 1 SUCCESS: 헤더 색상 원복 완료');
} else {
  console.log('PATCH 1 SKIP: 이미 원복되어 있거나 다른 상태');
}

// ── PATCH 2: 체크된(isDone) 진도 항목을 연한 파란색으로 ──────────
// 기존: background:var(--success-bg); border-color:var(--success);
// 변경: background:#eff6ff; border-color:#3b82f6;
const oldIsDone = "${isDone?'background:var(--success-bg); border-color:var(--success);':''}\" onclick=\"toggleStudentProgress";
const newIsDone = "${isDone?'background:#eff6ff; border-color:#3b82f6;':''}\" onclick=\"toggleStudentProgress";

if (content.includes(oldIsDone)) {
  content = content.replace(oldIsDone, newIsDone);
  console.log('PATCH 2 SUCCESS: 체크된 항목 배경색 → 연한 파란색');
} else {
  console.log('PATCH 2 FAILED: 대상 문자열을 찾지 못했습니다');
  process.exit(1);
}

// ── PATCH 3: 체크박스 checked 색상도 파란색으로 ────────────────
// hw-checkbox checked 클래스는 CSS에서 green으로 정의되어 있음
// 학생 진도 체크박스만 파란색으로: inline style 추가
const oldCheckbox = '              <div class="hw-checkbox ${isDone ? \'checked\' : \'\'}" style="width:20px; height:20px; flex-shrink:0; margin-top:2px"></div>';
const newCheckbox = '              <div class="hw-checkbox ${isDone ? \'checked\' : \'\'}" style="width:20px; height:20px; flex-shrink:0; margin-top:2px; ${isDone ? \'background:#3b82f6; border-color:#3b82f6;\' : \'\'}"></div>';

if (content.includes(oldCheckbox)) {
  content = content.replace(oldCheckbox, newCheckbox);
  console.log('PATCH 3 SUCCESS: 체크박스 파란색 적용');
} else {
  console.log('PATCH 3 FAILED: 체크박스 문자열 못찾음');
  // 진행은 계속 (없어도 기능상 무방)
}

fs.writeFileSync('index.html', content, 'utf8');
console.log('\n완료!');

File: fix_hw.py
import re

with open('index.html', 'r', encoding='utf-8') as f:
    content = f.read()

# Find the exact old block using a pattern
old_pattern = r"      const badgeClass = \(diff !== null && diff <= 2\) \? 'badge-danger' : 'badge-pending';\s+return `\s+<div class=\"admin-hw-item\".*?`;\s+}\)\.join\(''\);"

new_block = """      const badgeClass = (diff !== null && diff <= 2 && !chk) ? 'badge-danger' : 'badge-pending';
      const badgeText = chk ? '완료' : (d ? '취소 예정' : (diff !== null ? 'D-' + diff : '미완료'));
      const dateStr = h.createdAt || '';
      let monthStr = '미정';
      let monthColor = '#cbd5e1';
      if (dateStr.length >= 7) {
        const yyyyMm = dateStr.slice(0, 7);
        const yyyy = yyyyMm.slice(0, 4);
        const mm = parseInt(yyyyMm.slice(5, 7), 10);
        monthStr = yyyy + '년<br>' + mm + '월';
        let hash = 0;
        for (let i = 0; i < yyyyMm.length; i++) hash = yyyyMm.charCodeAt(i) + ((hash << 5) - hash);
        monthColor = 'hsl(' + (Math.abs(hash * 137) % 360) + ', 65%, 55%)';
      }
      return `
        <div class="detail-hw-item" style="padding:0; overflow:hidden; align-items:stretch; cursor:pointer; margin-bottom:8px; ${chk ? 'border-color:var(--success);' : ''}" onclick="toggleCheck('${h.id}',${d})">
          <div style="flex:0 0 70px; background:${monthColor}; color:white; display:flex; align-items:center; justify-content:center; text-align:center; font-size:11px; font-weight:700; line-height:1.4;">${monthStr}</div>
          <div style="flex:1; padding:12px 16px; display:flex; justify-content:space-between; align-items:center; ${chk ? 'background:var(--success-bg);' : ''}">
            <div class="detail-hw-info">
              <div class="detail-hw-title" style="${chk ? 'color:var(--text-muted)' : ''}">${h.title}</div>
              <div class="detail-hw-meta">${h.subject ? '<span class="cat-tag' + (h.subject.startsWith('수학') ? ' math' : '') + '" style="margin:0 6px 0 0; padding:2px 6px; font-size:10px">' + h.subject + '</span>' : ''}출제일: ${h.createdAt || '알수없음'} · <span style="color:var(--danger); font-weight:700">마감: ${h.dueDate || '미정'}</span></div>
            </div>
            <div class="detail-pct-wrap">
              <span class="detail-status hw-badge ${chk ? 'badge-done' : badgeClass}">${badgeText}</span>
              <div class="hw-checkbox ${chk ? 'checked' : ''}" style="width:22px; height:22px; margin-top:6px;"></div>
            </div>
          </div>
        </div>
      `;
    }).join('');"""

match = re.search(old_pattern, content, re.DOTALL)
if match:
    print(f"Found match at {match.start()}-{match.end()}")
    content = content[:match.start()] + new_block + content[match.end():]
    with open('index.html', 'w', encoding='utf-8') as f:
        f.write(content)
    print("Successfully updated!")
else:
    print("Pattern NOT found. Trying simpler search...")
    # Try line-by-line approach
    lines = content.split('\n')
    for i, line in enumerate(lines):
        if "const badgeClass = (diff !== null && diff <= 2) ? 'badge-danger' : 'badge-pending';" in line:
            print(f"Found at line {i+1}: {repr(line[:80])}")
            break
    else:
        print("Line not found either.")
